ui/src/store.js

```diff
@@ -102,7 +102,9 @@ export const {
     /** @type {(state: TreasuryState, v: { id: string, vault: VaultData }) => TreasuryState} */
     updateVault: ({ vaults, ...state }, { id, vault }) => {
       const oldVaultData = vaults && vaults[id];
-      const status = vault.liquidated ? VaultStatus.LIQUIDATED : vault.status;
+      const status = vault.liquidated
+        ? VaultStatus.LIQUIDATED
+        : (vault.status ?? oldVaultData?.status);
```

Comment:
why would `vault.status` not be defined? why is treating `oldVaultData` as current safe?

Reply:
The asset notifier updates the vault without a `status` because I wanted the vault notifier to be the source of truth for status instead. Without this change, the asset notifier will set `vault.status` to `undefined` every time it updates.

Reply:
ah. I see now you want the `vault.status` to override `oldVaultData.status` but only if it's defined.

Reply:
Yep exactly